File: 체육복.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
	int answer = 0;
	vector<int> uniform(n,1) ;
	for (int i = 0; i < lost.size(); i++) {
		uniform[lost[i] - 1] = 0;
	}
	for (int i = 0; i < reserve.size(); i++) {
		uniform[reserve[i] - 1]+=1;
	}


	for (int i = 0; i < n;i++) {
		if (uniform[i] == 2) {
			if (i == 0) {
				if(uniform[i + 1] == 0) {
					uniform[i] -= 1;
					uniform[i + 1] += 1;
					
				}
			}

			else if (i == n - 1) {
				if (uniform[i - 1] == 0) {
					uniform[i] -= 1;
					uniform[i - 1] += 1;
				}
			}
			else { 
					if (uniform[i - 1] == 0) {
						uniform[i] -= 1;
						uniform[i - 1] += 1;
					}else if (uniform[i + 1] == 0) {
						uniform[i] -= 1;
						uniform[i + 1] += 1;
					}
				}
			 
		}
		//cout << uniform[i] << " ";

	}

	for (int i = 0; i < n; i++) {
		if (uniform[i] != 0)
			answer++;
		//cout << uniform[i] << " ";

	}
	

	return answer;
}

void main() {
	int n = 5;
	vector<int> lost = {3,5};
	vector<int> reserve = {2,4};
	cout<<solution(n, lost, reserve);
}

File: 프로그래머스/level1/체육복.js
function solution (n, lost, reserve) {
    var answer = 0;
    const arr = Array(n).fill(1);
    lost.forEach(el => {
        arr[el - 1]--;
    })
    reserve.forEach(el => {
        arr[el - 1]++;
    })

    arr.forEach((el, i) => {
        if (el == 0) {
            if (i == 0) {//맨 앞
                if (arr[i + 1] == 2) {
                    arr[i + 1]--;
                    arr[i]++;
                }
            }
            else if (i == arr.length - 1) {
                if (arr[i - 1] == 2) {
                    arr[i - 1]--;
                    arr[i]++;
                }
            }
            else {
                if (arr[i + 1] == 2) {
                    arr[i + 1]--;
                    arr[i]++;
                } else if (arr[i - 1] == 2) {
                    arr[i - 1]--;
                    arr[i]++;

                }
            }
        }
    })


    return answer = arr.filter(el => el != 0).length;
}

console.log(solution(5, [2, 4], [3]));